src/tendencies/implicit/schur_complement_W.jl

```diff
@@ -302,15 +352,45 @@ function _ldiv_serial!(
         ∂ᶜ𝕋ₜ∂ᶠ𝕄 = getproperty(∂ᶜ𝕋ₜ∂ᶠ𝕄_field, ᶜ𝕋_name)
         @. xᶜ𝕋 = -bᶜ𝕋 + dtγ * apply(∂ᶜ𝕋ₜ∂ᶠ𝕄, xᶠ𝕄)
     end
-    for var_name in filter(is_edmf_var, propertynames(xc))
-        xᶜ𝕋 = getproperty(xc, var_name)
-        bᶜ𝕋 = getproperty(bc, var_name)
-        @. xᶜ𝕋 = -bᶜ𝕋
-    end
-    for var_name in filter(is_edmf_var, propertynames(xf))
-        xᶠ𝕋 = getproperty(xf, var_name)
-        bᶠ𝕋 = getproperty(bf, var_name)
-        @. xᶠ𝕋 = -bᶠ𝕋
+    if any(is_turbconv_var, propertynames(xc))
+        xᶜTC = xc.turbconv
+        xᶠTC = xf.turbconv
+        bᶜTC = bc.turbconv
+        bᶠTC = bf.turbconv
+        for var_prop_chain in Fields.property_chains(xᶜTC)
+            xᶜvar = Fields.single_field(xᶜTC, var_prop_chain, identity)
+            bᶜvar = Fields.single_field(bᶜTC, var_prop_chain, identity)
+            xᶜvar .= bᶜvar
+            xᶜvar_view = parent(xᶜvar)
+            ∂ᶜvarₜ∂ᶜvar =
+                Fields.single_field(∂ᶜTCₜ∂ᶜTC, var_prop_chain, identity)
+            @views ∂ᶜvarₜ∂ᶜvar_array = LinearAlgebra.Tridiagonal(
+                S_column_array.dl[1:(end - 1)],
+                S_column_array.d[1:(end - 1)],
+                S_column_array.du[1:(end - 1)],
```

Comment:
Is this properly viewed into? Above we're using `@views S_column_array.dl .= parent(S_column.coefs.:1)[2:end]`

Reply:
Ah, ok, right, `thomas_algorithm!` is populating the matrix